### DISCUSSION TITLE
install meshtasticd on Raspberry Pi Zero

### DISCUSSION THREAD

**Discussion by CrabbyPete** [meshtastic/firmware#3534] [Q&A] [answered]
I am try install the meshtasticd on my Raspberry Pi Zero and I keep getting the following message 

sudo apt install ./meshtasticd_2.2.18.e9bde80_arm64.deb
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'meshtasticd:arm64' instead of './meshtasticd_2.2.18.e9bde80_arm64.deb'
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 meshtasticd:arm64 : Depends: libyaml-cpp0.7:arm64 but it is not installable
E: Unable to correct problems, you have held broken packages.
(lora) pete@raspberry:~ $ sudo apt install libyaml-cpp0.7:arm64
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libyaml-cpp0.7:arm64
E: Couldn't find any package by glob 'libyaml-cpp0.7'

I am using the latest bookwork version of RaspianOS

**Comment by jp-bennett** [comment]
> Depends: libyaml-cpp0.7:arm64

Is this an arm32 system? That would be my guess. I'm hoping to get arm32 building as part of CI before too much longer, but not there yet. For now, if this is a Pi zero 2, you can install the 64 bit Bookworm. For an original Pi zero, it takes a 32 bit compile.

**Comment by ON2ON-HAM** [comment]
I use the Raspi 3 B+ with this operating system : 

Static hostname: raspberrypi
       Icon name: computer
      Machine ID: 162b36b6acf044df8689f65959406588
         Boot ID: b5addceb509842abb53aab72d0700b96
Operating System: Debian GNU/Linux 12 (bookworm)
          Kernel: Linux 6.6.20+rpt-rpi-v8
    Architecture: arm64

No problems on installing MeshtasticD, and using my DIY Lora Hat.

**Comment by stephenbrad515** [comment]
I am having the same issue as CrabbyPete, I believe is am using a 64 bit hardware and OS.

uname -a
Linux lorapie 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux

cat /proc/cpuinfo
Hardware        : BCM2835
Revision        : a02082
Serial          : 000000000c17bc2f
Model           : Raspberry Pi 3 Model B Rev 1.2

cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian

sudo apt install ./meshtasticd_2.3.2.63df972_arm64.deb 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'meshtasticd' instead of './meshtasticd_2.3.2.63df972_arm64.deb'
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 meshtasticd : Depends: libyaml-cpp0.7 but it is not installable
E: Unable to correct problems, you have held broken packages.

udo apt install ./libyaml-cpp0.7_0.7.0+dfsg-8+b1_arm64.deb 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libyaml-cpp0.7' instead of './libyaml-cpp0.7_0.7.0+dfsg-8+b1_arm64.deb'
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 libyaml-cpp0.7 : Depends: libc6 (>= 2.32) but 2.31-13+rpt2+rpi1+deb11u8 is to be installed
                  Depends: libstdc++6 (>= 11) but 10.2.1-6 is to be installed

**Reply by markbirss** [reply, marked as answer]
You running unsupported OS, switch to Debian 12 Bookworm 64bit for Raspberry Pi

**Reply by stephenbrad515** [reply]
Your suggestion worked, thank you.
